sdk/src/main/scala/com/horizen/account/network/SyncStatusActor.scala

```diff
@@ -0,0 +1,227 @@
+    context.system.eventStream.subscribe(self, classOf[GetHighestBlockTimestamp])
+    context.system.eventStream.subscribe(self, classOf[SemanticallySuccessfulModifier[PMOD]])
+  }
+
+  override def postStop(): Unit = {
+    log.debug("SidechainBlock Actor is stopping...")
+    super.postStop()
+  }
+
+  def retrieveCurrentBlock(sidechainNodeView: View): (PMOD) = {
+    sidechainNodeView.history.getBestBlock
+  }
+
+  def retrieveCurrentHeight(sidechainNodeView: View): (Int) = {
+    sidechainNodeView.history.getCurrentHeight
+  }
+
+  // retrieve the timestamp from the block recently applied to the state
+  // if the difference between the highestBlockTimestamp and the timestamp of the block applied is less than
+  // the block rate multiplied for a confidence parameter we set the sync status to false.
+  // This means that the node is synched with the rest of the network.
+  def processNewBlockApplied(sidechainBlock: PMOD): Unit = {
+    if((highestBlockTimestamp - sidechainBlock.timestamp).compareTo(params.consensusSlotsInEpoch * confidenceParameter) < 0) {
+      syncStatus = false
+      // reset the internal state values to null
+      startingBlock = 0; highestBlock = 0; highestBlockTimestamp = 0L
+      // broadcast sync stop
+      context.system.eventStream.publish(NotifySyncStop(false))
+    }
+  }
+
+  // calculate the block correction parameter retrieving the last n blocks (n = previousBlocksRange) and check how many slot were occupied
+  // if the previousBlocksRange value is less than the number of blocks present in the node history we just return the value 1
+  // this previous case can be encounter at node first startup without blocks in history
+  def calculateBlockCorrection(sidechainNodeView: View): (Double) = {
+    val currentBlockHeight = sidechainNodeView.history.getCurrentHeight
+    if(previousBlocksRange.compare(currentBlockHeight)<0) {
+      val currentBlockId = sidechainNodeView.history.getBlockIdByHeight(currentBlockHeight).get()
+      val currentBlock = sidechainNodeView.history.getBlockById(currentBlockId).get()
+      val pastBlockId = sidechainNodeView.history.getBlockIdByHeight(currentBlockHeight - previousBlocksRange).get()
+      val pastBlock = sidechainNodeView.history.getBlockById(pastBlockId).get()
+      val timestampDifference = currentBlock.timestamp - pastBlock.timestamp
+      val expectedBlocksInRange = timestampDifference / params.consensusSecondsInSlot.toDouble
+      val slotCorrection = previousBlocksRange / expectedBlocksInRange
+      slotCorrection
+    } else 1
+  }
+
+  // process the new highest timestamp from the batch of new modifiers received by an other peer
+  def processNewHighestBlockTimestamp(highestBlockTimestampFromMessage: Long): Unit = {
+
+    // retrieve from the view the current block and current block height
+    val currentBlockFromView = Await.result(sidechainNodeViewHolderRef ? GetDataFromCurrentView(retrieveCurrentBlock), timeoutDuration)
```

Comment:
general recomendations:
1) keep as less requests to the `sidechainNodeViewHolderRef` as it is possible - they cost time
2) avoid getting the whole block (because we look into the database), instead rely on blockInfo (it is cached, much smaller, contains timestamp as well).

Reply:
1. with the current implementation the requests to the `sidechainNodeViewHolderRef` are done only when the `syncStatus` is _False_ and a new block applied to the state, from the second block we just update the internal fields.
2. implementation changed to retrieve the `blockInfo` instead of the whole block